src/test/java/com/google/sps/scheduler/SchedulerTest.java

```diff
@@ -532,19 +605,139 @@ public void scheduleInRangeWhenStartDateEqualsEndDate() {
             createDateTime(day, 14, 0, ZURICH_TIME_ZONE)));
 
     Assert.assertEquals(expectedScheduledTasks, actualScheduledTasks);
   }
+
+  @Test
+  public void fitSingleWorkhourslong() {
+    // Events :
+    // Day    : |-------------------------------|
+    // Tasks  : |-------------------------------|
+    List<Event> calendarEvents = Collections.emptyList();
+    LocalDate day = LocalDate.of(2048, 6, 23);
+
+    Scheduler scheduler = new Scheduler(calendarEvents, varyingDurationSample, ZURICH_TIME_ZONE);
+    List<ExtendedTask> actualScheduledTasks = scheduler.scheduleInRange(day, day);
+    List<ExtendedTask> expectedScheduledTasks = Arrays.asList(
+        createCustomDurationTaskWithDue(createDateTime(day, Scheduler.START_HOUR, Scheduler.START_MINUTE, ZURICH_TIME_ZONE), WORKING_HOURS)
+    );
+
+    Assert.assertEquals(expectedScheduledTasks, actualScheduledTasks);
+  }
+
+  @Test
+  public void varyingDurations() {
+    // Events :    |-A|  |-B--|               |-C---|
+    // Day    : |----------------------------------------|
+    // Tasks  : |5m|  |1h|    |-2h-|30m|10m|5m|     |-2h-|
+    List<Event> calendarEvents = new ArrayList<>();
+    LocalDate day = LocalDate.of(2048, 6, 23);
+    Event eventA = createEvent(
+        createDateTime(day, Scheduler.START_HOUR, Scheduler.START_MINUTE + 5, ZURICH_TIME_ZONE),
+        createDateTime(day, 10, 0, ZURICH_TIME_ZONE),
+        ZURICH_TIME_ZONE);
+    Event eventB = createEvent(
+        createDateTime(day, 11, 0, ZURICH_TIME_ZONE),
+        createDateTime(day, 12, 0, ZURICH_TIME_ZONE),
+        ZURICH_TIME_ZONE);
+    Event eventC = createEvent(
+        createDateTime(day, 15, 0, ZURICH_TIME_ZONE),
+        createDateTime(day, 16, 0, ZURICH_TIME_ZONE),
+        ZURICH_TIME_ZONE);
+
+    calendarEvents.add(eventA);
+    calendarEvents.add(eventB);
+    calendarEvents.add(eventC);
+
+    Scheduler scheduler = new Scheduler(calendarEvents, varyingDurationSample, ZURICH_TIME_ZONE);
+    List<ExtendedTask> actualScheduledTasks = scheduler.scheduleInRange(day, day);
+    List<ExtendedTask> expectedScheduledTasks = Arrays.asList(
+        createCustomDurationTaskWithDue(
+            createDateTime(day,Scheduler.START_HOUR, Scheduler.START_MINUTE, ZURICH_TIME_ZONE), FIVE_MINS),
+        createCustomDurationTaskWithDue(
+            createDateTime(day, 10, 0, ZURICH_TIME_ZONE), ONE_HOUR),
+        createCustomDurationTaskWithDue(
+            createDateTime(day, 12, 0, ZURICH_TIME_ZONE), TWO_HOURS),
+        createCustomDurationTaskWithDue(
+            createDateTime(day, 14, 0, ZURICH_TIME_ZONE), HALF_AN_HOUR),
+        createCustomDurationTaskWithDue(
+            createDateTime(day, 14, 30, ZURICH_TIME_ZONE), TEN_MINS),
+        createCustomDurationTaskWithDue(
+            createDateTime(day, 16, 0, ZURICH_TIME_ZONE), TWO_HOURS));
+
+    Assert.assertEquals(expectedScheduledTasks, actualScheduledTasks);
+  }
+
+  @Test
+  public void outOfTasks() {
```

Comment:
If testing "out of tasks", please again use "far in the future" end date for scheduling

Reply:
Right.

Reply:
Thanks in general for going through the test cases thoroughly. :)